velox/dwio/text/writer/TextWriter.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/dwio/text/writer/TextWriter.h"
+#include "velox/common/base/Pointers.h"
+#include "velox/exec/MemoryReclaimer.h"
+
+namespace facebook::velox::text {
+
+static std::string getColumnData(
+    const std::shared_ptr<DecodedVector>& columnDecodedData,
+    const TypePtr& type,
+    vector_size_t row) {
+  if (columnDecodedData->isNullAt(row)) {
+    return NULLDATA;
+  }
+
+  switch (type->kind()) {
+    case TypeKind::BOOLEAN:
+      return std::to_string(columnDecodedData->valueAt<bool>(row));
+    case TypeKind::TINYINT:
+      return std::to_string(columnDecodedData->valueAt<int8_t>(row));
+    case TypeKind::SMALLINT:
+      return std::to_string(columnDecodedData->valueAt<int16_t>(row));
+    case TypeKind::INTEGER:
+      return std::to_string(columnDecodedData->valueAt<int32_t>(row));
+    case TypeKind::BIGINT:
+      return std::to_string(columnDecodedData->valueAt<int64_t>(row));
+    case TypeKind::REAL:
+      return std::to_string(columnDecodedData->valueAt<float>(row));
+    case TypeKind::DOUBLE:
+      return std::to_string(columnDecodedData->valueAt<double>(row));
+    case TypeKind::VARCHAR:
+      return columnDecodedData->valueAt<StringView>(row).getString();
+    case TypeKind::TIMESTAMP:
```

Comment:
We don't need the data copy here and we could provide a bunch of template methods to accelerate